server/datastore/mysql/mysql.go

```diff
@@ -314,42 +315,126 @@ func (d *Datastore) MigrateData(ctx context.Context) error {
 	return data.MigrationClient.Up(d.writer.DB, "")
 }
 
-func (d *Datastore) MigrationStatus(ctx context.Context) (fleet.MigrationStatus, error) {
+type migrationRecord struct {
+	VersionID int64 `db:"version_id"`
+	IsApplied bool  `db:"is_applied"`
+}
+
+// loadMigrations manually loads the applied migrations in ascending
+// order (goose doesn't provide such functionality).
+func (d *Datastore) loadMigrations(
+	ctx context.Context,
+) (tableRecs []migrationRecord, dataRecs []migrationRecord, err error) {
+	// We need to run the following to trigger the creation of the migration status tables.
+	tables.MigrationClient.GetDBVersion(d.writer.DB)
+	data.MigrationClient.GetDBVersion(d.writer.DB)
+	// version_id > 0 to skip the bootstrap migration that creates the migration tables.
+	if err := sqlx.SelectContext(ctx, d.reader, &tableRecs,
+		"SELECT version_id, is_applied FROM "+tables.MigrationClient.TableName+" WHERE version_id > 0 ORDER BY id ASC",
```

Comment:
`IsApplied` in the struct is not used, and we probably want to ignore migrations that are recorded but not applied (not sure when that happens, tbh). So maybe remove it from the struct and do `WHERE ... AND is_applied`?

Reply:
Good catch!, fixed.